scripts/commit-msg

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set -o errexit
+
+exit_() {
+    echo ""
+    echo "$1"
+    echo ""
+    echo "This hook can be skipped if needed with 'git commit --no-verify'"
+    echo "See '.git/hooks/commit-msg', installed from 'scripts/commit-msg'"
+    exit 1
+}
+
+sign_offs="$(grep '^Signed-off-by: ' "$1" || test $? = 1 )"
+
+if [ -z "$sign_offs" ]; then
+    exit_ "Commit failed: please sign-off on the DCO with 'git commit -s'"
+fi
+
+if [ -n "$(echo "$sign_offs" | sort | uniq -c | sed -e '/^[ 	]*1[ 	]/d')" ]; then
```

Comment:
What is this black magic?

Reply:
It's part of the git commit-msg.sample that I carried forward. It takes all the found 'Signed-off-by:' lines found by the grep, sorts them, enumerates all unique values (`uniq -c`) and removes all lines with only a count of 1. The `sed` command just makes finding the lines prefixed by the count of `1` robust against differences in whitespaces.